src/api/pipeline/client.mli

```diff
@@ -0,0 +1,44 @@
+open Capnp_rpc_lwt
+
+module Build_status : sig
+  type t = Raw.Reader.BuildStatus.t
+
+  val pp : t Fmt.t
+  val to_string : t -> string
+end
+
+module State : sig
+  type t = Raw.Reader.JobInfo.State.unnamed_union_t
+
+  val pp : t Fmt.t
+  val from_build_status : [< `Failed | `Not_started | `Passed | `Pending ] -> t
```

Comment:
This doesn't match up with the variant used for `BuildStatus.t` it really should.
I would suggest using a Poly Variant for BuildStatus.t and referring to it here. It isn't strictly necessary but it helps when reading the types to see they are talking about the same thing.